code/modelCuration/getCompFromUniprotCellAtlas.py

```diff
@@ -0,0 +1,138 @@
+# collect subcellular localization for existing enzymes from Uniprot/Swissprot and Cell Atlas
+# compartment info from both sources were combined and integrated into genes.tsv
+
+import pandas as pd
+import csv
+
+# import subcellular location from SwissProt annotation
+SwissProt_tsv = pd.read_table("~/Downloads/SwissProt_20221115.tsv")
+SwissProt_proteins = SwissProt_tsv['Entry'].to_list()
+SwissProt_subCellularlocation = SwissProt_tsv['Subcellular location [CC]'].to_list()
+
+
+# build a dict for key word mapping of compartment assignment
+swissprot_keywords = {
+    #'Extracellular': ['',''],   # e
+    'Peroxisome': ['peroxisome'],   # x
+    'Mitochondria': ['mitochondrion {','mitochondrion matrix'],   # m
+    'Cytosol': ['cytoplasm','cytosol','cytoskeleton'],   # c
+    'Lysosome': ['lysosome','lysosome lumen','lysosome membrane'],   # l
+    'Endoplasmic reticulum': ['endoplasmic reticulum','Endoplasmic reticulum lumen','endoplasmic reticulum membrane'],   # r
+    'Golgi apparatus': ['golgi apparatus','golgi apparatus lumen','golgi apparatus membrane'],   # g
+    'Nucleus': ['nucleus','nucleolus'],   # n
+    'Inner mitochondria': ['mitochondrion inner membrane','mitochondrion intermembrane space']    # i
+}
+
+
+# store swissport compartment info as a dict with a key of uniprot id
+comps_from_swissprot = {}
+for i, text in enumerate(SwissProt_subCellularlocation):
+    out_list = []  # output compartmetns as a list
+    comps_from_swissprot[SwissProt_proteins[i]] = out_list
+    if not pd.isna(text):
+        for key in swissprot_keywords:
+            # check if cellular location text contains any keyword from a compartment
+            if any(word in text.lower() for word in swissprot_keywords[key]):
+                out_list.append(key)
+        comps_from_swissprot[SwissProt_proteins[i]] = out_list
+
+
+# load Human-GEM gene annotation file
+HumanGenes_tsv = pd.read_table("~/Documents/GitHub/Human-GEM/model/genes.tsv")
+Human_genes = HumanGenes_tsv['genes'].to_list()
+Human_proteins = HumanGenes_tsv['geneUniProtID'].to_list()
+
+
+# save SwissProt compartment information to tsv file
+swissprot_compList = ['']*len(Human_genes)
+for i in range(len(Human_genes)):
+    if not pd.isna(Human_proteins[i]):
+        if Human_proteins[i] in SwissProt_proteins:
+            swissprot_compList[i] = ';'.join(comps_from_swissprot[Human_proteins[i]])
```

Comment:
Is that possible also extract the info of wether the protein is a membrane protein?

Reply:
yes, there's curated info about membrane proteion